File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/di/AppComponent.java
package ru.avystavkin.githubgists.di;

import javax.inject.Singleton;

import dagger.Component;
import ru.avystavkin.githubgists.screen.main.MainPageActivity;
import ru.avystavkin.githubgists.screen.gist_detail.GistDetailActivity;
import ru.avystavkin.githubgists.screen.user_detail.UserDetailActivity;

@Singleton
@Component(modules = {DataModule.class})
public interface AppComponent {

    void injectGistActivity(MainPageActivity activity);

    void injectGistDetailActivity(GistDetailActivity activity);

    void injectUserDetailActivity(UserDetailActivity activity);
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/di/AppComponent.kt
package ru.avystavkin.githubgists.di

import dagger.Component
import ru.avystavkin.githubgists.screen.gist_detail.GistDetailActivity
import ru.avystavkin.githubgists.screen.main.MainPageActivity
import ru.avystavkin.githubgists.screen.user_detail.UserDetailActivity
import javax.inject.Singleton

@Singleton
@Component(modules = [DataModule::class])
interface AppComponent {

    fun injectGistActivity(activity: MainPageActivity)

    fun injectGistDetailActivity(activity: GistDetailActivity)

    fun injectUserDetailActivity(activity: UserDetailActivity)
}
